scalding-core/src/test/scala/com/twitter/scalding/ExecutionTest.scala

```diff
@@ -417,5 +415,15 @@ class ExecutionTest extends WordSpec with Matchers {
 
       assert(result.waitFor(Config.default, Local(true)).get == 0.to(10).reverse)
     }
+
+    "can hashCode, compare, and run a long sequence" in {
+      val execution = Execution.sequence((1 to 100000).toList.map(Execution.from(_)))
```

Comment:
Yeah the breaking point varied (and could change within the life of a process), but hits rather consistently.

Against clean develop, all actually die on `hashCode`:
```
[info] Exception encountered when attempting to run a suite with class name: com.twitter.scalding.ExecutionTest *** ABORTED ***
[info]   java.lang.StackOverflowError:
[info]   at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
[info]   at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:168)
[info]   at com.twitter.scalding.Execution$Zipped.hashCode(Execution.scala:547)
[info]   at scala.runtime.ScalaRunTime$.hash(ScalaRunTime.scala:206)
[info]   at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
[info]   at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
[info]   at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:168)
[info]   at com.twitter.scalding.Execution$Mapped.hashCode(Execution.scala:406)
[info]   at scala.runtime.ScalaRunTime$.hash(ScalaRunTime.scala:206)
[info]   at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
[info]   at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
[info]   at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:168)
[info]   at com.twitter.scalding.Execution$Zipped.hashCode(Execution.scala:547)
[info]   at scala.runtime.ScalaRunTime$.hash(ScalaRunTime.scala:206)
[info]   at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
[info]   at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
[info]   at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:168)
[info]   at com.twitter.scalding.Execution$Mapped.hashCode(Execution.scala:406)
[info]   at scala.runtime.ScalaRunTime$.hash(ScalaRunTime.scala:206)
[info]   at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
[info]   at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
[info]   at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:168)
[info]   at com.twitter.scalding.Execution$Zipped.hashCode(Execution.scala:547)
[info]   at scala.runtime.ScalaRunTime$.hash(ScalaRunTime.scala:206)
[info]   at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
[info]   at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
[info]   at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:168)
[info]   at com.twitter.scalding.Execution$Mapped.hashCode(Execution.scala:406)
...
```

If we patch just that part, we see this on the `run`:
```
[info] Exception encountered when attempting to run a suite with class name: com.twitter.scalding.ExecutionTest *** ABORTED ***
[info]   java.lang.StackOverflowError:
[info]   at scala.runtime.ScalaRunTime$.hash(ScalaRunTime.scala:206)
[info]   at scala.util.hashing.MurmurHash3$$anonfun$unorderedHash$1.apply(MurmurHash3.scala:92)
[info]   at scala.util.hashing.MurmurHash3$$anonfun$unorderedHash$1.apply(MurmurHash3.scala:91)
[info]   at scala.collection.immutable.HashMap$HashMap1.foreach(HashMap.scala:221)
[info]   at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:428)
[info]   at scala.util.hashing.MurmurHash3.unorderedHash(MurmurHash3.scala:91)
[info]   at scala.util.hashing.MurmurHash3$.mapHash(MurmurHash3.scala:222)
[info]   at scala.collection.GenMapLike$class.hashCode(GenMapLike.scala:35)
[info]   at scala.collection.AbstractMap.hashCode(Map.scala:59)
[info]   at com.twitter.scalding.Config$class.hashCode(Config.scala:376)
[info]   at com.twitter.scalding.Config$$anon$1.hashCode(Config.scala:458)
[info]   at scala.runtime.ScalaRunTime$.hash(ScalaRunTime.scala:206)
[info]   at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
[info]   at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
[info]   at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:168)
[info]   at scala.Tuple2.hashCode(Tuple2.scala:20)
[info]   at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1012)
[info]   at java.util.concurrent.ConcurrentHashMap.putIfAbsent(ConcurrentHashMap.java:1535)
[info]   at com.twitter.scalding.FutureCache.getOrPromise(FutureCache.scala:40)
[info]   at com.twitter.scalding.FutureCache.getOrElseUpdateIsNew(FutureCache.scala:21)
[info]   at com.twitter.scalding.Execution$EvalCache.getOrElseInsertWithFeedback(Execution.scala:379)
[info]   at com.twitter.scalding.Execution$EvalCache.getOrElseInsert(Execution.scala:384)
[info]   at com.twitter.scalding.Execution$FutureConst.runStats(Execution.scala:388)
[info]   at com.twitter.scalding.Execution$Zipped$$anonfun$runStats$9.apply(Execution.scala:552)
[info]   at com.twitter.scalding.Execution$Zipped$$anonfun$runStats$9.apply(Execution.scala:551)
[info]   at com.twitter.scalding.FutureCache.getOrElseUpdateIsNew(FutureCache.scala:24)
[info]   at com.twitter.scalding.Execution$EvalCache.getOrElseInsertWithFeedback(Execution.scala:379)
[info]   at com.twitter.scalding.Execution$EvalCache.getOrElseInsert(Execution.scala:384)
[info]   at com.twitter.scalding.Execution$Zipped.runStats(Execution.scala:551)
...
```